### PR TITLE
Run tests in CI, and fix fonts for snapshot testing

### PR DESCRIPTION
**Pull request by DJMcNab**
See also linebender/vello#439

This should open the door to doing much more comprehensive testing at higher level in the nearish future (including e.g. in `xilem_masonry`)

### CODE REVIEW

**Comment by DJMcNab** [comment]
Discussion in [#masonry > Snapshot test differences in Masonry CI](https://xi.zulipchat.com/#narrow/stream/317477-masonry/topic/Snapshot.20test.20differences.20in.20Masonry.20CI)

File: crates/masonry/src/testing/harness.rs
Comment (PoignardAzur) [inline crates/masonry/src/testing/harness.rs:192]: Wouldn't that path be wrong for any dependent crate creating a test harness?

I think you want to use `include_bytes` instead (until we come up with an actual resource loading system).

File: crates/masonry/src/render_root.rs
Comment (PoignardAzur) [inline crates/masonry/src/render_root.rs:166]: Seems reasonable. RenderRoot is the type ultimately responsible for turning semantic data into bitmaps. You can either pass a test font with a constructor argument, a build method, or a setter. I think a setter is fine.

**Comment by PoignardAzur** [comment]
Will try to run this on my machine this afternoon, then review.

**Comment by DJMcNab** [comment]
Ooh, a time limit to get it working? Exciting!

**Comment by DJMcNab** [comment]
The status of this is that there's a lot more work needed on testing, including moving this into Xilem. However, this does get our existing tests working in a cross-platform way.

One concern I have with the size of screenshot tests. I'm planning on raising this in office hours, to see if anyone has any good solutions, so won't merge until them.
Ideally, we could just use git lfs, but they don't want to support

**Comment by PoignardAzur** [comment]
Unit tests pass on my machine! :tada: 

(PopOs 22.04, RTX 2070 with X11, for what it's worth.)

**Comment by DJMcNab** [comment]
I'm going to postpone merging this PR until after I port https://github.com/linebender/vello/pull/647 to this repository, to avoid adding the new snapshots into main as-is.

I think I'll do that in a different PR, because it also has implications on other caching choices, and shouldn't be controversial.

**Comment by PoignardAzur** [comment]
Makes sense, yes. FWIW the developer experience of git-lfs in Vello seems pretty good to me.

File: masonry/src/widget/tests/safety_rails.rs
Comment (PoignardAzur) [inline masonry/src/widget/tests/safety_rails.rs:83]: This is incorrect: you can run unit tests in release mode.

This should probably be an early exit instead.

**Review by PoignardAzur — APPROVED** [review]
LGTM.

File: masonry/src/widget/tests/safety_rails.rs
Comment (DJMcNab) [reply masonry/src/widget/tests/safety_rails.rs:83]: An early exit isn't possible here, because this is a `should_panic` test.

Comment (PoignardAzur) [reply masonry/src/widget/tests/safety_rails.rs:83]: Oooh! That's annoying. I guess then we should either:

- `#[cfg]` the test.
- Create a synthetic panic in release mode.
- Use `catch_unwind` instead.

In any case, you don't need to solve this now, but a TODO comment would be nice.

Comment (PoignardAzur) [reply masonry/src/widget/tests/safety_rails.rs:83]: Created #477 for this, you can link to it.

File: masonry/src/testing/harness.rs
Comment (DJMcNab) [reply masonry/src/testing/harness.rs:534]: I'll do that when I rebase this past #475

File: masonry/src/testing/harness.rs
Comment (DJMcNab) [reply masonry/src/testing/harness.rs:544]: I'm not sure. I'm wanting to redo a lot of this testing code, but I think that will want to come after linebender/rfcs#7

**Comment by DJMcNab** [comment]
Requested review just to get final sign-off on the latest pushed changed (since ac7549d8ade33d95ef925737c4aab5ed156c2861)

File: masonry/src/testing/harness.rs
Comment (PoignardAzur) [reply masonry/src/testing/harness.rs:539]: Wait, no, I misread the code. We already redraw in the early exit, the only thing this skips is actually sending the Vello scene to GPU. So that makes sense.

File: masonry/src/widget/tests/safety_rails.rs
Comment (PoignardAzur) [inline masonry/src/widget/tests/safety_rails.rs:59]: I don't think you need to link to the issue. Doing a `cfg_attr` fixes it.

For the message I'd write something like:

"Safety rails are disabled when debug_assertions is false (i.e. in release mode)."

**Review by PoignardAzur — APPROVED** [review]
LGTM. Tests still compile on my machine.

**Pull request merged**
